src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -188,8 +189,26 @@ object WdlExpression {
             }
           case _ => Failure(new WdlExpressionException("Right-hand side of expression must be identifier"))
         }
+      case a: Ast if a.isArrayOrMapIndex =>
```

Comment:
I found this name a little confusing. The index is actually just the rhs of this element. Maybe ArrayOrMapLookup?